### PR TITLE
Extend the model to automatically perform input normalization and output normalization if exported in onnx

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
As per title, this PR embeds three linear layers that perform the normalization of the input and the denormalization of the network's output. 

This feature is useful since we do not need to copy the normalization parameters in a separate file since they will be automatically stored in the model used to perform the inference. This will mitigate what @S-Dafarra commented in https://github.com/ami-iit/bipedal-locomotion-framework/pull/652#discussion_r1169692905

For the time being, this feature is implemented only for onnx but still, we may think to:
1. make it optional in onnx 
2. move the feature directly in the PyTorch training script and generate a model that doesn't require to perform normalization of the input to be used

### CODE REVIEW

File: scripts/convert_to_onnx.py [hunk 1]
```diff
@@ -1,22 +1,55 @@
...
     mann_restored = torch.load(str(model_path))
 
     # Set dropout and batch normalization layers to evaluation mode before running inference
     mann_restored.eval()
+    input_size = next(mann_restored.parameters()).size()[1]
+
+    # Here we create two layes for normalization and denormalization
+    X_mean = read_from_file(str(normalization_folder / "X_mean.txt"))
+    X_std = read_from_file(str(normalization_folder / "X_std.txt"))
+    X_std[np.where(X_std <= 1e-4)] = 1
```
Comment (paolo-viceconte) [inline scripts/convert_to_onnx.py:20]: This is how I was neglecting zero variances up to now, but after reading @S-Dafarra's comment [here](https://github.com/ami-iit/bipedal-locomotion-framework/pull/652#discussion_r1169822799) I guess that using the  threshold (1e-4 here) is actually more appropriate.

We should update the normalization also at training time though, to use it consistently at inference time.

File: scripts/convert_to_onnx.py [hunk 2]
```diff
@@ -1,22 +1,55 @@
...
+
+    Y_mean = read_from_file(str(normalization_folder / "Y_mean.txt"))
+    Y_std = read_from_file(str(normalization_folder / "Y_std.txt"))
+    Y_std[np.where(Y_std <= 1e-4)] = 1
```
Comment (paolo-viceconte) [inline scripts/convert_to_onnx.py:24]: Same comment here

**Review by paolo-viceconte — APPROVED** [review]
Apart from the minor comments on how to neglect zero variances, looks good to me!

File: scripts/convert_to_onnx.py (hunk 1 above)
Comment (paolo-viceconte) [reply scripts/convert_to_onnx.py:20]: > We should update the normalization also at training time though, to use it consistently at inference time.

Let's check this at the next training attempt, but merge this PR in the meanwhile. We will integrate these changes in a new PR if needed.
